src/Symfony/Component/Notifier/Bridge/LightSms/LightSmsTransport.php

```
@@ -0,0 +1,151 @@
+    {
+        $this->login = $login;
+        $this->password = $password;
+        $this->phone = $phone;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('lightsms://%s?phone=%s', $this->getEndpoint(), $this->phone);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $signature = $this->generateSignature($message);
+
+        $endpoint = sprintf(
+            'https://%s/external/get/send.php?login=%s&signature=%s&phone=%s&text=%s&sender=%s&timestamp=%s',
+            $this->getEndpoint(),
+            $this->login,
+            $signature,
+            str_replace('+', '', $message->getPhone()),
+            $message->getSubject(),
+            $this->phone,
+            time()
+        );
+
+        $response = $this->client->request('GET', $endpoint);
+
+        $content = $response->toArray(false);
+
+        if (isset($content['error'])) {
+            throw new TransportException('Unable to send the SMS: '.self::ERROR_CODES[$content['error']], $response);
+        }
+
+        $phone = preg_replace("/[^\d]/", '', $message->getPhone());
+        if (32 == $content[$phone]['error']) {
+            throw new TransportException('Unable to send the SMS: '.self::ERROR_CODES[$content['error']], $response);
+        }
+
+        if (0 == $content[$phone]['error']) {
+            $sentMessage = new SentMessage($message, (string) $this);
+            $sentMessage->setMessageId($content[$phone]['id_sms']);
+        }
+
+        return $sentMessage;
```

this variable is undefined/null when `$content[$phone]['error']` !== 0 and 32

---

I always have here variable in array
```
LightSmsTransport.php on line 131:
array:7 [▼
  "error" => "0"
  "id_sms" => "6170129423143422190002"
  "cost" => 0.019
  "count_sms" => 1
  "sender" => "Kuro-Kainos"
  "network" => null
  "ported" => false
]
```

---

you can not guarantee that error will be always `0`. IMHO you should throw an exception in such a situation and build the `sentMessage` only when error is 0

---

I think this one will help me

```
 $response = $this->client->request('GET', $endpoint);

        if (Response::HTTP_OK !== $response->getStatusCode()) {
            throw new TransportException('Unable to send the SMS: ', $response);
        }
```

---

this is not related. 
you check only for error code == 32, and then build an object only when code is 0.

this is fragile and does not handled all the other cases.